driver.sh

```diff
@@ -139,14 +140,51 @@ setup_variables() {
 }
 
 check_dependencies() {
+  # Check for existence of needed binaries
   command -v nproc
   command -v "${CROSS_COMPILE:-}"as
   command -v ${qemu}
   command -v timeout
   command -v unbuffer
-  command -v clang-9
-  command -v llvm-ar-9
-  command -v "${LD:="${CROSS_COMPILE:-}"ld}"
+
+  # Check for LD, CC, and AR environmental variables
+  # and print the version string of each. If CC and AR
+  # don't exist, try to find them.
+  # lld isn't ready for all architectures so it's just
+  # simpler to fall back to GNU ld when LD isn't specified
+  # to avoid architecture specific selection logic.
+
+  "${LD:="${CROSS_COMPILE:-}"ld}" --version
+
+  if [[ -z "${CC:-}" ]]; then
+    for CC in clang-9 clang-8 clang-7 clang; do
+      ${CC} --version 2>/dev/null && break
```

Comment:
what happens when no version is found?

Reply:
Hmmm fair point, I'll need to add another check to make sure we fail here instead of during make.